### ISSUE TITLE
[Feature request] add openCL acceleration support?

### ISSUE THREAD

**Issue by skidd-level-100** [ggerganov/llama.cpp#1447]
I want (https://github.com/ggerganov/llama.cpp/commit/905d87b70aa189623d500a28602d7a3a755a4769) but with openCL,
plz

off topic:
also why would an integrated gpu on a laptop worked fine with openCL and then after git pull + recomp fail with -6 openCL error?

this is very weird to me

its an AMD intergated (radeon)

**Comment by SlyEcho** [comment]
The accelerated generation is also being worked on by @0cc4m: #1459 
The error may be related to some changes that I made recently, there is a fix coming in #1435

**Comment by AutonomicPerfectionist** [comment]
>  -6 openCL error

@skidd-level-100 I found that just using the fix from #1435 was not enough for me. I also needed to export these shell variables (on Linux):
```bash
export GPU_FORCE_64BIT_PTR=1
export GPU_USE_SYNC_OBJECTS=1
export GPU_MAX_ALLOC_PERCENT=100
```
